src/btreemap.rs

```diff
@@ -20,7 +20,7 @@ const MAGIC: &[u8; 3] = b"BTR";
 ///
 /// The implementation is based on the algorithm outlined in "Introduction to Algorithms"
 /// by Cormen et al.
-pub struct BTreeMap<M: Memory, K: Storable, V: Storable> {
+pub struct BTreeMap<K: BoundedStorable, V: BoundedStorable, M: Memory> {
```

Comment:
The use of `Storable` here, and not `BoundedStorable`, is deliberate (ditto everywhere else), since users can still use the `BTreeMap` with unbounded `Storable` types. It's a flaw in our test suite that this change didn't break any tests.

Reply:
That's a fair point. I am currently exploring removing the bound altogether, but it's not clear whether or not this exploration will be fruitful, so, until then, it would make sense to make using `BoundedStorable` required, as it does capture the intent better than specifying the sizes in the constructor. We should remove the `init_with_sizes` API in this case though, but that I'm happy to do in a follow-up PR.